config/routes.rb

```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  get '/users', to: 'users#index', as: 'users'
+  get '/users/:user_id', to: 'users#show', as: 'user'
```

Comment:
- [ ] On line no. 5, it should be `id` instead of `user_id` as per project requirements.

Also, I would recommend writing a URL like the one I have shown below in a RESTFUL way so that you don't have to write all the URLs one by one.

```ruby
# config/routes.rb

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
@Dipeshtwis Great suggestions, I have a question what if I want to add `:user_id` or `posts_id` to this URL while having this approach how can I add it?

Reply:
As you know, rails follows convention over configuration. If you want to do it you have to it manually the way you are doing right now.